hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/keyvalue/impl/TestBlockManagerImpl.java

```diff
@@ -196,6 +198,52 @@ public void testPutAndGetBlock(ContainerTestVersionInfo versionInfo)
 
   }
 
+  @ContainerTestVersionInfo.ContainerTest
+  public void testPutBlockForClosed(ContainerTestVersionInfo versionInfo)
+      throws Exception {
+    initTest(versionInfo);
+    assertEquals(0, keyValueContainer.getContainerData().getBlockCount());
```

Comment:
`assertEquals(State.CLOSED, keyValueContainer.getContainerData().getState());` fails. The container should be closed to test the method.

Reply:
The container state is checked one function above in the `KeyValueHandler`. `BlockManager` handles the metadata updates.